.github/workflows/integration.yml

```diff
@@ -12,71 +12,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Run secret-dependent integration tests only after /ok-to-test approval
-on:
-  pull_request:
-  repository_dispatch:
-    types: [ok-to-test-command]
-
+# Run secret-dependent integration tests only after approval
 name: Integration tests
-env:
-  IMAGE_NAME: scorecard
-  OUTPUT_PATH: output
+on: pull_request_target
 
 permissions:
   contents: read
   pull-requests: write
 
 jobs:
-  integration-trusted:
+  approve:
     runs-on: ubuntu-latest
     steps:
-      - id: debug-outputs
-        run: |
-          echo ${{ github.event_name }}
-          echo ${{ github.event.client_payload }}
-
-      - id: verify-sha-input
-        if: github.event_name == 'repository_dispatch'
-        run: |
-          echo \"${{ github.event.client_payload.pull_request.head.sha }}\"
-          echo \"${{ github.event.client_payload.slash_command.args.named.sha }}\"
-          SHAINPUT=$(echo ${{github.event.client_payload.slash_command.args.named.sha}} | cut -c1-7)
-          if [ ${#SHAINPUT} -le 6 ]; then echo "error::input sha not at least 7 characters long" ; exit 1
-          else echo "done"
-          fi
-          SHAHEAD=$(echo ${{github.event.client_payload.pull_request.head.sha}} | cut -c1-7)
-          echo ${#SHAINPUT}
-          echo ${#SHAHEAD}
-          if [ "${SHAHEAD}" != "${SHAINPUT}" ]; then echo "sha input from slash command does not equal the head sha" ; exit 1
-          else echo "shas are equal"
-          fi
-
-      - name: pull_request actions/checkout
-        if: github.event_name == 'pull_request'
-        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.3.4
+      - name: approve
+        run: echo For security reasons, all pull requests need to be approved before running integration tests.
 
+  integration-trusted:
+    runs-on: ubuntu-latest
+    environment: integration-test
+    needs: [approve]
```

Comment:
doesn't the approve job always succeed? confused what approval is given

Reply:
I should probably name that job better. `approve` job here just prints a message (I was testing whether `integration` workflow actually runs). The real approval comes from the `environment: integration-test` line. That is a protected environment which can only be accessed by PRs that have approval from one of the required approvers.